# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/AudioTrack.cpp
# Commit: c54cb3c10992
# Full Hash: c54cb3c109922bc15fb6d0ca67e5f2a05980c1a5
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 2
# Description:
#   Bug 1493613 - Move MediaStream control from DOMMediaStream to MediaStreamTrack. r=padenot
#   
#   This is inherently large, because modifying these bits of DOMMediaStream and
#   MediaStreamTrack affects all consumers and producers of all DOMMediaStreams and
#   MediaStreamTracks.
# ==============================================================================

diff -r 0b03dd9d20ac -r c54cb3c10992 dom/media/AudioTrack.cpp
--- a/dom/media/AudioTrack.cpp	Wed Jul 31 09:35:06 2019 +0000
+++ b/dom/media/AudioTrack.cpp	Wed Jul 31 07:58:17 2019 +0000
@@ -14,9 +14,20 @@
 
 AudioTrack::AudioTrack(nsIGlobalObject* aOwnerGlobal, const nsAString& aId,
                        const nsAString& aKind, const nsAString& aLabel,
-                       const nsAString& aLanguage, bool aEnabled)
+                       const nsAString& aLanguage, bool aEnabled,
+                       AudioStreamTrack* aStreamTrack)
     : MediaTrack(aOwnerGlobal, aId, aKind, aLabel, aLanguage),
-      mEnabled(aEnabled) {}
+      mEnabled(aEnabled),
+      mAudioStreamTrack(aStreamTrack) {}
+
+AudioTrack::~AudioTrack() = default;
+
+NS_IMPL_CYCLE_COLLECTION_INHERITED(AudioTrack, MediaTrack, mAudioStreamTrack)
+
+NS_IMPL_ADDREF_INHERITED(AudioTrack, MediaTrack)
+NS_IMPL_RELEASE_INHERITED(AudioTrack, MediaTrack)
+NS_INTERFACE_MAP_BEGIN_CYCLE_COLLECTION(AudioTrack)
+NS_INTERFACE_MAP_END_INHERITING(MediaTrack)
 
 JSObject* AudioTrack::WrapObject(JSContext* aCx,
                                  JS::Handle<JSObject*> aGivenProto) {